docs/4.x/plugin-directory-structure.md

```diff
@@ -0,0 +1,54 @@
+---
+category: Develop
+title: Plugin Directory Structure
+---
+# Matomo Plugin Directory Structure
+
+Each plugin has its own directory within the root `plugins/` directory. Generally, plugins have a similar [structure as the Matomo core](/guides/how-piwik-works) for most things. Most plugins have only few files, while some others have more files. Below is a list of some files and directories that may exist in a plugin.
+
+
+| File / Directory      | Description |
+| ----------- | ----------- |
+| plugin.json      | Contains [metadata about the plugin](/guides/distributing-your-plugin#prepare-your-plugin) such as contact information, a description, and more. Mostly needed for plugins on the Marketplace.      |
+|  $PluginName.php  |    `$PluginName` needs to be replaced with the name of the plugin. This file is the plugin file which allows plugins to listen to [events](/guides/events) and method hooks such as `install()`, `activate` and `uninstall()`.      |
+|  API.php  |    Defines [API methods](/guides/expose-api-methods) for this plugin.     |
+|  Archiver.php  |    This file includes logic on how to create archive reports from log raw data.     |
+|  Categories/  |     This directory includes menu categories and subcategories that are used to build the reporting menu. They define for example the order in which they are shown in the menu.    |
+|  Columns/ |     This directory includes [dimensions](/guides/dimensions) and metrics.    |
+|  Commands/ |     This directory includes [console commands](/guides/piwik-on-the-command-line).    |
+|  config/ |     This directory can include various files like "config.php" and "test.php" to change Matomo using [dependency injection](/guides/dependency-injection).    |
+|  Controller.php  |     Defines [controller actions](/guides/pages) for this plugin.    |
+|  Dao/  |     This directory includes data access objects that connect mostly to the database and execute queries. It doesn't contain much logic usually.    |
+|  DataTable/Filter  |     This directory may include [custom plugin data table filters](/guides/datatable#custom-filter).    |
+|  Diagnostic/  |     This directory typically includes [system checks](/guides/system-check).    |
```

Comment:
There might also be `LocationProvider/`, which can include additional geolocation providers.